### PR TITLE
Abort on packaging errors

### PR DESCRIPTION
**Pull request by noxasaxon**
Abort serverless deploy when plugin fails to package a lambda function properly.

if abortOnPackagingErrors: true - 
- pip install git clone will print to stderr, ignore it 
- check if error is related to docker, if so abort
- all other errors, prompt to continue or abort

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -111,17 +112,51 @@ class PkgPyFuncs {
...
+      console.log(noText)
+      this.error('Aborting')
+      return
+    }
+  }
+
   setupContainer(){
     let out = this.runProcess('docker',['ps', '-a', '--filter',`name=${this.containerName}`,'--format','{{.Names}}'])
     out = out.replace(/^\s+|\s+$/g, '')
+    console.log('SDLFKJSDL:KFJSDL:KFJ ', this.containerName)
```
Comment (ubaniabalogun) [inline index.js]: I believe this is no longer needed

File: index.js [hunk 2]
```diff
@@ -111,17 +112,51 @@ class PkgPyFuncs {
...
+        this.requestUserConfirmation()
+      }
+
     }
 
-    const out = ret.stdout.toString()
     return out
   }
 
+  requestUserConfirmation(prompt="\n\n??? Do you wish to continue deployment with the stated errors? \n",
```
Comment (ubaniabalogun) [inline index.js]: Let's put the `abort on error` feature behind a configuration flag. This change may break the config of someone who is cool with the errors

File: index.js (hunk 1 above)
Comment (noxasaxon) [reply index.js]: lol!

File: README.md [hunk 3]
```diff
@@ -133,6 +133,7 @@ The plugin configurations are simple:
...
 | dockerImage        | The Docker image to use to compile functions if `useDocker` is set to `true`. Must be specified as `repository:tag`. If the image doesn't exist on the system, it will be downloaded. The initial download may take some time.                            | Yes. Defaults to `lambci/lambda:build-${provider.runtime}` |
 | containerName      | The desired name for the Docker container.                                                                                                                                                                         | Yes. Defaults to `serverless-package-python-functions`     |
+| usePackagingErrorAlerts | Boolean indicating whether you want to stop deployment when packaging errors are detected. Example errors include: `useDocker` is enabled but the Docker service is not running, pip finds dependency mismatches, virtual environment errrors, etc.. When an error is detected, this will prompt via commandline to continue or abort deploy. | Yes. Defaults to `false` |
```
Comment (ubaniabalogun) [inline README.md]: Let's go with a more descriptive/intuitive name, Like abortOnPackagingErrors

File: README.md [hunk 4]
```diff
@@ -133,6 +133,7 @@ The plugin configurations are simple:
...
 | dockerImage        | The Docker image to use to compile functions if `useDocker` is set to `true`. Must be specified as `repository:tag`. If the image doesn't exist on the system, it will be downloaded. The initial download may take some time.                            | Yes. Defaults to `lambci/lambda:build-${provider.runtime}` |
 | containerName      | The desired name for the Docker container.                                                                                                                                                                         | Yes. Defaults to `serverless-package-python-functions`     |
+| abortOnPackagingErrors | Boolean indicating whether you want to stop deployment when packaging errors are detected. Example errors include: `useDocker` is enabled but the Docker service is not running, pip finds dependency mismatches, virtual environment errrors, etc.. When an error is detected, this will prompt via commandline to continue or abort deploy. | Yes. Defaults to `false` |
```
Comment (ubaniabalogun) [inline README.md]: Instead of `Example errors include:` let's go with: `Examples of scenarios that will cause packaging errors include:`

**Pull request merged**